WebRTCPlugin/RTCPeerConnection.cpp

```diff
@@ -2,480 +2,481 @@
-  if (track.vt != VT_DISPATCH)
-    return E_INVALIDARG;
-
-  IDispatch* disp = V_DISPATCH(&track);
-
-  if (!disp)
-    return E_INVALIDARG;
-
-  //Get atl com object from track.
-  CComPtr<ITrackAccess> proxy;
-  HRESULT hr = disp->QueryInterface(IID_PPV_ARGS(&proxy));
-  if (FAILED(hr))
-    return hr;
-
-  //libwebrtc only supports adding track to one stream and it only requires the stream label, 
-  //so we use a fake object instead of having to wrap the whole media stream 
-  std::vector<webrtc::MediaStreamInterface*> streams;
-  rtc::scoped_refptr<webrtc::MediaStreamInterface> mediaStream;
-  std::string label = (char*)_bstr_t(stream);
-  //If attaching to a stream
-  if (!label.empty())
-  {
-    auto it = localStreams.find(label);
-	  //Check if we already have it
-    if (it == localStreams.end())
-    {
-      //Create new fake stream
-      mediaStream = new FakeMediaStream(label);
-      //Add to list of local streams
-      localStreams[label] = mediaStream;
-    } else {
-      //Get it
-      mediaStream = it->second;
-    }
-    //Add it to list
-    streams.push_back(mediaStream.get());
-  }
-  
-  //Add track
-  rtc::scoped_refptr<webrtc::RtpSenderInterface > senderInterface = pc->AddTrack(proxy->GetTrack(),streams);
-
-  //Create activeX object for media stream track
-  CComObject<RTPSender>* sender;
-  HRESULT hresult = CComObject<RTPSender>::CreateInstance(&sender);
-
-  if (FAILED(hresult))
-	  return hresult;
-
-  //Attach to native object
-  sender->Attach(senderInterface);
-
-  //Get Reference to pass it to JS
-  *rtpSender = sender->GetUnknown();
-
-  //Add JS reference
-  (*rtpSender)->AddRef();
-
-  //Done
-  return S_OK;
+	if (!pc)
+		return E_UNEXPECTED;
+
+	//Get dispatch interface
+	if (track.vt != VT_DISPATCH)
+		return E_INVALIDARG;
+
+	IDispatch* disp = V_DISPATCH(&track);
+
+	if (!disp)
+		return E_INVALIDARG;
+
+	//Get atl com object from track.
+	CComPtr<ITrackAccess> proxy;
+	HRESULT hr = disp->QueryInterface(IID_PPV_ARGS(&proxy));
+	if (FAILED(hr))
+		return hr;
+
+	// libwebrtc only supports adding track to one stream and it only requires the stream label, 
+	// so we use a fake object instead of having to wrap the whole media stream 
+	//std::vector<webrtc::MediaStreamInterface*> streams;
+	std::vector<std::string> streams;
+	rtc::scoped_refptr<webrtc::MediaStreamInterface> mediaStream;
+	std::string id = (char*)_bstr_t(stream);
+	//If attaching to a stream
+	if (!id.empty())
+	{
+		auto it = localStreams.find(id);
+		//Check if we already have it
+		if (it == localStreams.end())
+		{
+			//Create new fake stream
+			mediaStream = new FakeMediaStream(id);
+			//Add to list of local streams
+			localStreams[id] = mediaStream;
+		}
+		else {
+			//Get it
+			mediaStream = it->second;
+		}
+		//Add it to list
+		streams.push_back(mediaStream.get()->id());
+	}
+
+	//Add track
+	rtc::scoped_refptr<webrtc::RtpSenderInterface > senderInterface = pc->AddTrack(proxy->GetTrack(), streams).value();

```

Comment:
what will happen if failed

Reply:
If you mean that we should check the return value of AddTrack before using .value(), then that's a good idea.